matchtree.go

```diff
@@ -196,6 +196,8 @@ type symbolRegexpMatchTree struct {
 
 func (t *symbolRegexpMatchTree) prepare(doc uint32) {
```

Comment:
Maybe we could rename `prepare` to `advance`. That's a much more familiar concept for an iterator, and makes it clear it's important to move forward the iterator's position (which affects `nextDoc`).

Reply:
Thanks for the context here and the deep dive into `nextFileMatch`! It does seem we need a bigger refactor/ fix ... renames can just be part of that to clarify the new semantics.